aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -0,0 +1,351 @@
+    String nextToken = null;
+    List<TableIdentifier> results = new ArrayList<>();
+    do {
+      GetTablesResponse response = glue.getTables(GetTablesRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.toDatabaseName(namespace))
+          .nextToken(nextToken)
+          .build());
+      if (response.hasTableList()) {
+        results.addAll(response.tableList().stream()
+            .map(GlueToIcebergConverter::toTableId)
+            .collect(Collectors.toList()));
+        nextToken = response.nextToken();
+      }
+    } while (nextToken != null);
+    LOG.debug("Listing of namespace: {} resulted in the following tables: {}", namespace, results);
+    return results;
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    try {
+      TableOperations ops = newTableOps(identifier);
+      TableMetadata lastMetadata = ops.current() != null ? ops.current() : null;
+      glue.deleteTable(DeleteTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.getDatabaseName(identifier))
+          .name(identifier.name())
+          .build());
+      LOG.info("Successfully dropped table {} from Glue", identifier);
+      if (purge && lastMetadata != null) {
+        CatalogUtil.dropTableData(ops.io(), lastMetadata);
+        LOG.info("Glue table {} data purged", identifier);
+      }
+      LOG.info("Dropped table: {}", identifier);
+      return true;
+    } catch (EntityNotFoundException e) {
+      LOG.error("Cannot drop table {} because table not found or not accessible", identifier, e);
+      return false;
+    } catch (Exception e) {
+      LOG.error("Cannot complete drop table operation for {} due to unexpected exception", identifier, e);
+      throw e;
+    }
+  }
+
+  /**
+   * Rename table in Glue is a drop table and create table.
+   * @param from identifier of the table to rename
+   * @param to new table name
+   */
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    // check new namespace exists
+    if (!namespaceExists(to.namespace())) {
+      throw new NoSuchNamespaceException("Cannot rename %s to %s because namespace %s does not exist",
+          from, to, to.namespace());
+    }
+    // keep metadata
+    Table fromTable = null;
+    try {
+      GetTableResponse response = glue.getTable(GetTableRequest.builder()
+          .catalogId(catalogId)
+          .databaseName(IcebergToGlueConverter.getDatabaseName(from))
+          .name(IcebergToGlueConverter.getTableName(from))
+          .build());
+      fromTable = response.table();
+    } catch (EntityNotFoundException e) {
+      throw new NoSuchTableException(e, "Cannot rename %s because the table does not exist in Glue", from);
+    }
+    dropTable(from, false);
+    // use the same Glue info to create the new table, pointing to the old metadata
+    glue.createTable(CreateTableRequest.builder()
```

Comment:
Sorry for coming back to this again, I wonder if it's worth adding a try catch block here and try recreating the old `fromTable` if this throws, so that for validation errors we didn't detect earlier or transient glue 5xx error we still could restore the table to the old state

Reply:
sounds good, let me do that